File: MNG/00/MNGDay00.py
"""Marches And Gnatts - Puzzle 0
Solution Started: Jul 25, 2025
Puzzle Link: https://mng.quest/quest/tutorial
Solution by: Abbas Moosajee
Brief: [Binary Increment]
"""

#!/usr/bin/env python3

import os, re, copy, time, sys

turing_path = os.path.join(os.path.dirname(__file__), "..", "TuringMachine")
sys.path.append(turing_path)
from TuringMachine import MachineLogic, TuringConfig, TuringMachine

init_rules = """
    INIT | FIND | R
    FIND | FIND | R
    FIND _ HALT | R
    """
init_tape = "||||"

if __name__ == "__main__":
    _, results, used_resources = TuringMachine(init_rules).run_machine(init_tape, play_type = 0, visualize=True)
    print_info = results + [""] + used_resources
    print("\n".join(print_info))

File: MNG/00/MNGDay00.js
/**
 * Marches And Gnatts - Puzzle 0
Solution Started: Jul 25, 2025
Puzzle Link: https://mng.quest/quest/tutorial
Solution by: Abbas Moosajee
Brief: [Binary Increment]
 */
import { MNG_ARENA } from '../MNG_ARENA.js';

const unary_inc_rules = `
INIT | FIND | R
FIND | FIND | R
FIND _ HALT | R
`;

const unary_inc = new MNG_ARENA("unary_increment")
unary_inc.full_solution(unary_inc_rules);
unary_inc.benchmark_solution(["||||", "||", "|||"], false)
// unary_inc.test_solution("||||", true)
// unary_inc.save_rules(import.meta.url);

